### ISSUE TITLE
nb-coverage failing for coverage 5.x

### ISSUE THREAD

**Issue by likewei92** [chrisjsewell/pytest-notebook#6]
When running `--nb-coverage` with coverage 5.x, I got the error
```
pytest_notebook.execution.CoverageError: An error occurred while executing coverage teardown:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-13-929d854e3f33> in <module>
      3 # __cov.save()
      4 __stream = __StringIO()
----> 5 __cov.get_data().write_fileobj(__stream)
      6 print(__stream.getvalue())

AttributeError: 'CoverageData' object has no attribute 'write_fileobj'
```
According to the coverage docs, `write_fileobj` is indeed gone: https://coverage.readthedocs.io/en/coverage-5.0.3/api_coveragedata.html

I downgraded coverage to 4.5.4, but I still get an error:
```
pytest_notebook.execution.CoverageError: The teardown coverage cell did not produce the expected output: [{'output_type': 'stream', 'name': 'stdout', 'text': '!coverage.py: This is a private format, don\'t read it directly!{"lines":{}}'}, {'output_type': 'stream', 'name': 'stdout', 'text': '\n'}]
```

**Comment by chrisjsewell** [comment]
Hey thanks for the feedback, and sorry for the late response 😬 . I might not be able to get round to fixing this in the immediate future, but am happy to receive/merge PRs

**Comment by AdamGleave** [comment]
I looked into this briefly to see if it made sense to fix in another PR, wanted to add a quick summary of the problem in case anyone wants to work on it in the future:
  - Coverage 5.x now uses a SQLData format for coverage compared to an in-house format in 4.5.x. The tl;dr is they're completely different formats, and there is no equivalent to `write_fileobj` that I can see.
  - `CoverageData` does support a `dumps()` format, but it's proprietary.
  - We don't seem to work with raw coverage data very many places, though -- main thing is in the tests, which could probably be redesigned. So it may be viable to serialize with `dumps()`, deserialize with `loads()` and then interact with the `CoverageData` API for e.g. coverage merging.

**Comment by chrisjsewell** [comment]
Thanks for looking into this @AdamGleave

**Comment by chrisjsewell** [comment]
Now fixed in pytest-notebook 0.7.0
